sdks/java/core/src/main/java/org/apache/beam/sdk/io/CompressedSource.java

```diff
@@ -380,18 +395,77 @@ public T getCurrent() throws NoSuchElementException {
       return readerDelegate.getCurrent();
     }
 
+    @Override
+    public final long getParallelismConsumed() {
+      if (splittable) {
+        return readerDelegate.getParallelismConsumed();
+      }
```

Comment:
Alternatively, for methods that are "here is the splittable implementation and here is the non-splittable implementation" use an "if/else" block to make that clear.